macros/models/_project/balancer/balancer_bpt_prices_macro.sql

```diff
@@ -0,0 +1,220 @@
+            SUM(amount) OVER (PARTITION BY pool_id, token ORDER BY DAY ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS cumulative_amount
+        FROM daily_delta_balance
+    ),
+
+    calendar AS (
+        SELECT date_sequence AS day
+        FROM unnest(sequence(date('2021-04-21'), date(now()), interval '1' day)) as t(date_sequence)
+    ),
+
+   cumulative_usd_balance AS (
+        SELECT
+            c.day,
+            '{{blockchain}}' as blockchain,
+            b.pool_id,
+            b.token,
+            cumulative_amount / POWER(10, COALESCE(t.decimals, p1.decimals)) * COALESCE(p1.price, 0) AS protocol_liquidity_usd
+        FROM calendar c
+        LEFT JOIN cumulative_balance b ON b.day <= c.day
+        AND c.day < b.day_of_next_change
+        LEFT JOIN {{ source('tokens', 'erc20') }} t ON t.contract_address = b.token
+        AND blockchain = '{{blockchain}}'
+        LEFT JOIN prices p1 ON p1.day = b.day
+        AND p1.token = b.token
+        WHERE b.token != BYTEARRAY_SUBSTRING(b.pool_id, 1, 20)
+    ),
+
+    weighted_pool_liquidity_estimates AS (
+        SELECT
+            b.day,
+            b.pool_id,
+            q.name,
+            pool_type,
+            ROW_NUMBER() OVER (partition by b.day, b.pool_id ORDER BY SUM(b.protocol_liquidity_usd) ASC) AS pricing_count, --to avoid double count in pools with multiple pricing assets
+            SUM(b.protocol_liquidity_usd) / COALESCE(SUM(w.normalized_weight), 1) AS protocol_liquidity
+        FROM cumulative_usd_balance b
+        LEFT JOIN {{ ref('balancer_pools_tokens_weights') }} w ON b.pool_id = w.pool_id 
+        AND b.token = w.token_address
+        AND b.protocol_liquidity_usd > 0
+        LEFT JOIN {{ ref('balancer_token_whitelist') }} q ON b.token = q.address 
+        AND b.blockchain = q.chain
+        LEFT JOIN pool_labels p ON p.pool_id = BYTEARRAY_SUBSTRING(b.pool_id, 1, 20)
+        WHERE q.name IS NOT NULL 
+        AND p.pool_type IN ('WP', 'WP2T') -- filters for weighted pools with pricing assets
+        AND w.blockchain = '{{blockchain}}'
+        GROUP BY 1, 2, 3, 4
+    ),
+    
+    weighted_pool_liquidity_estimates_2 AS(
+    SELECT  e.day,
+            e.pool_id,
+            SUM(e.protocol_liquidity) / MAX(e.pricing_count) AS protocol_liquidity
+    FROM weighted_pool_liquidity_estimates e
+    GROUP BY 1,2
+    ),
+    
+    tvl AS(
+    SELECT
+        c.day,
+        BYTEARRAY_SUBSTRING(c.pool_id, 1, 20) AS pool_address,
+        '2' AS version,
+        '{{blockchain}}' AS blockchain,
+        SUM(COALESCE(b.protocol_liquidity * w.normalized_weight, c.protocol_liquidity_usd)) AS liquidity
+    FROM cumulative_usd_balance c
+    FULL OUTER JOIN weighted_pool_liquidity_estimates_2 b ON c.day = b.day
+    AND c.pool_id = b.pool_id
+    LEFT JOIN {{ ref('balancer_pools_tokens_weights') }} w ON b.pool_id = w.pool_id 
+    AND w.blockchain = '{{blockchain}}'
+    AND w.token_address = c.token
+    LEFT JOIN eth_prices e ON e.day = c.day 
```

Comment:
You're missing the `eth_prices` CTE definition here. I was able to run `balancer_v2_arbitrum_bpt_prices_beta` locally with this CTE:

```
eth_prices as (
            select date_trunc('day', minute) as day, avg(price) as eth_price
            from {{ source("prices", "usd") }}
            where symbol = 'ETH'
            group by 1
        ),
```

Reply:
thank you so much, @couralex6!!!

Reply:
great find @couralex6 🙏